### PR TITLE
Preprocessor checks to build on CPUs

### PR DESCRIPTION
**Pull request by itopcuoglu**

### CODE REVIEW

**Comment by itopcuoglu** [comment]
@marchdf

**Review by marchdf — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -53,7 +55,10 @@ if (ENABLE_HIP)
     hip::host)
 endif()
 
-
+if(ENABLE_HIP OR ENABLE_CUDA)
+  set(USE_GPU ON)
```
Comment (jrood-nrel) [inline CMakeLists.txt:59]: This line can be removed.

File: CMakeLists.txt [hunk 2]
```diff
@@ -7,6 +7,8 @@ set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
 option(ENABLE_CUDA "Enable Nvidia CUDA" OFF)
 option(ENABLE_HIP "Enable HIP/ROCm" OFF)
 option(ENABLE_UMPIRE "Enable Umpire GPU memory pools" OFF)
+option(USE_GPU "Check if building on GPUs" OFF)
```
Comment (jrood-nrel) [inline CMakeLists.txt:10]: Both of these options can be removed.

File: CMakeLists.txt (hunk 1 above)
Comment (itopcuoglu) [reply CMakeLists.txt]: Addressed this.

File: CMakeLists.txt (hunk 2 above)
Comment (itopcuoglu) [reply CMakeLists.txt]: I also addressed this.

File: CMakeLists.txt [hunk 3]
```diff
@@ -7,6 +7,7 @@ set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
 option(ENABLE_CUDA "Enable Nvidia CUDA" OFF)
 option(ENABLE_HIP "Enable HIP/ROCm" OFF)
 option(ENABLE_UMPIRE "Enable Umpire GPU memory pools" OFF)
+option(USE_APPLE_CLANG "Check if building on MacOS " OFF)
```
[the remark this answers is not in the document]
Comment (itopcuoglu) [reply CMakeLists.txt]: Removed.

**Review by jrood-nrel — APPROVED** [review]

**Pull request merged**